week4/solutions.js

```diff
@@ -0,0 +1,166 @@
+const task2_data = [
+  { id: 1, score: 90 },
+  { id: 2, score: 40 },
+  { id: 3, score: 80 }
+];
+R.adjust(
+  R.evolve({
+    score: R.add(10)
+  }),
+  R.findIndex(R.propEq("id", 2))(task2_data)
+)(task2_data);
+
+/* #########################################################################
+ Task3:
+     input: [1, 2, 3] 
+     output: [<div key={0}>1</div>, <div key={1}>2</div>, <div key={2}>3</div>]
+############################################################################ */
+var mapIndexed = R.addIndex(R.map);
+mapIndexed((val, idx) => "<div key=" + idx + ">" + idx + "</div>", [1, 2, 3]);
+
+/* #########################################################################
+ Task4: 
+     inpput: {name: "John", age: 20} 
+     output: [<div key="name">John</div>, <div key="age">20</div>]
+############################################################################ */
+var transforma = (num, key, obj) => "<div key=" + "key" + ">" + num + "</div>";
+R.values(R.mapObjIndexed(transforma, { name: "John", age: 20 }));
+
+/* #########################################################################
+ Task5:
+    input: [{id: 1}, {id: 2}, {id: 3}]
+    output: to Just({id: 2})
+############################################################################ */
+const task5_data = [{ id: 1 }, { id: 2 }, { id: 3 }];
+R.compose(
+  Maybe.fromNullable,
+  R.find(R.propEq("id", 2))
+)(task5_data);
+
+/* #########################################################################
+ Task6:
+    input: 
+      const tasks = {
+        "tasks": [{
+          "id": 6441,
+          "name": "Teach me how to fly a drone",
+          "slug": "drone"
+        }]
+      }
+      const activities = [{taskId: 6441}, {taskId: 289}]
+    output:
+      [Just({"id": 6441, "name": "Teach me how to fly a drone", "slug": "drone"}), Nothing]
+############################################################################ */
+const task6_tasks = {
+  tasks: [
+    {
+      id: 6441,
+      name: "Teach me how to fly a drone",
+      slug: "drone"
+    }
+  ]
+};
+const activities = [{ taskId: 6441 }, { taskId: 289 }];
+
+const findBytaskId = data => id => {
+  return R.compose(
+    Maybe.fromNullable,
+    R.find(R.propEq("id", id))
+  )(data);
+};
+
+console.log(
+  R.compose(
+    R.map(
+      R.compose(
+        findBytaskId(task6_tasks.tasks),
+        R.prop("taskId")
+      )
+    )
+  )(activities)
+);
+
+/* #########################################################################
+ Task7:
+      input: 
+         const payload = {payloadString: '20'}
+         const scores = [{score: 10}, {score: 15}]
+      output:
+        [{score: 20}, {score: 10}, {score: 15}]
+############################################################################ */
+const task7_payload = { payloadString: "20" };
+const task7_scores = [{ score: 10 }, { score: 15 }];
+R.compose(
+  R.prepend(R.__, task7_scores),
+  R.objOf("score"),
+  a => +a
+)(task7_payload);
```

Comment:
May need to pass `task7_payload. payloadString` instead of `task7_payload`? :)

Reply:
ye